src/python/pants/source/source_root.py

```diff
@@ -19,6 +20,10 @@ class SourceRootCategories:
 
 SourceRoot = namedtuple('_SourceRoot', ['path', 'langs', 'category'])
 
+# Named tuple of path/langs/category, where the langs are deliberately not canonicalized in order to match up
+# with the actual directory names on disk
+UncanonicalizedSourceRoot = namedtuple('UncanonicalizedSourceRoot', ['path', 'langs', 'category'])
```

Comment:
Type hints version is even better :) https://docs.python.org/3/library/typing.html#typing.NamedTuple